### PR TITLE
Fix the loop fuser tests on the host

### PR DESCRIPTION
**Pull request by adayton1**
- Add toss clang host config
- Fix build warnings (e.g. unused variables, deprecated behavior for implicitly defined copy constructors)
- Rename LoopFuserTest.cxx to TestLoopFuser.cxx to be consistent with all other test file names
- Fix memory leaks in loop fuser tests
- Fix loop fuser tests on the host

### CODE REVIEW

**Review by liu15 — APPROVED** [review]

File: test/TestLoopFuser.cxx [hunk 1]
```diff
@@ -143,7 +135,8 @@ GPU_TEST(TestPacker, packFixedRangeMacro) {
 
    care::gpuDeviceSynchronize(__FILE__, __LINE__);
 
-   // pack should  not have happened yet so
+#ifdef CARE_GPUCC
```
Comment (dtaller) [inline test/TestLoopFuser.cxx:138]: Maybe add comment by CARE_GPUCC protection is needed here

Comment (dtaller) [reply test/TestLoopFuser.cxx:138]: Maybe add comment by CARE_GPUCC protection is needed here and also below in other places where you did this, just for clarity

**Review by dtaller — APPROVED** [review]
In the Loop Fuser Test file, just add a comment on why #ifdef CARE_GPUCC was needed in the places you added it, just for clarity.

besides that, looks good!

**Review by robinson96 — APPROVED** [review]
LGTM

File: test/TestLoopFuser.cxx (hunk 1 above)
Comment (adayton1) [reply test/TestLoopFuser.cxx:138]: Updated the comment (now it specifically mentions the device, so I think that should be sufficient).

**Pull request merged**